### PR TITLE
[JENKINS-58051] Hook for plugins not using standard tags

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
[JENKINS-58051](https://issues.jenkins-ci.org/browse/JENKINS-58051) This PR introduces a hook that can be extended trivially by plugins not following the standard tag conventions in Jenkins to make the PCT able to checkout them from Github

### CODE REVIEW

**Comment by raul-arabaolaza** [comment]
@oleg-nenashev @batmat added Unit and IT tests

**Comment by raul-arabaolaza** [comment]
Legitimate test failure in windows, fixing it.

**Comment by raul-arabaolaza** [comment]
And fixed and ready for review

File: plugins-compat-tester/src/main/resources/nonstandardtagplugins.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+electricflow=cloudbees-flow-%s
+electricflow-minimumVersion=1.1.8
```
Comment (MRamonLeon) [inline plugins-compat-tester/src/main/resources/nonstandardtagplugins.properties]: Is not more interesting to have the maximunVersion? If a plugin changes, it is to become compliant with the rules, so the latest versions most likely, don't need the hook anymore. Or maybe both properties. WDYT? 

BTW: I was working in exactly the same hook but specifically in a job on our CI, so :100:

**Pull request merged**